### PR TITLE
Dynamically negotiate the more recent data stream version

### PR DESCRIPTION
**Pull request by pasnox**
Task-Id: KDEND-161

### CODE REVIEW

File: common/message.cpp [hunk 1]
```diff
@@ -175,6 +175,45 @@ Message Message::readMessage(QIODevice *device)
...
 }
 
+quint8 Message::lowestSupportedDataVersion()
+{
+    return QDataStream::Qt_4_8;
+}
+
+quint8 Message::highestSupportedDataVersion()
+{
+#if QT_VERSION >= QT_VERSION_CHECK(5, 0, 0)
```
Comment (krf) [inline common/message.cpp]: I think we can just return `Qt_DefaultCompiledVersion` here.

From qdatastream.h:
```
#if QT_VERSION >= 0x050900
#error Add the datastream version for this Qt version and update Qt_DefaultCompiledVersion
#endif
        Qt_DefaultCompiledVersion = Qt_5_8
```

File: client/client.h [hunk 2]
```diff
@@ -94,12 +94,13 @@ private slots:
...
 private:
     enum InitState {
-        VersionChecked = 1,
-        ObjectMapReceived = 2,
-        ServerInfoReceived = 4,
-        ConnectionEstablished = 8,
+        VersionChecked = 0x1,
+        ObjectMapReceived = 0x2,
+        ServerInfoReceived = 0x4,
+        ServerDataVersionNegociated = 0x8,
```
Comment (krf) [inline client/client.h]: Typo: `negociated`

Comment (pasnox) [inline client/client.h]: done

File: common/message.cpp (hunk 1 above)
Comment (pasnox) [inline common/message.cpp]: Ah did not known about this one, thanks.

**Pull request merged**